stdlib/REPL/src/REPLCompletions.jl

```diff
@@ -811,4 +811,13 @@ function shell_completions(string, pos)
     return Completion[], 0:-1, false
 end
 
+Base.Experimental.register_error_hint(UndefVarError) do io, ex
```

Comment:
Any reason to put this into the REPL stdlib, whereas e.g. the noncallable number hint is in base? Either seems fine, but I think it would be good to be consistent about this.

Reply:
I want it to be unavailable if REPL isn't loaded, since that lets us do more advanced things here (though I should perhaps move it to `__init__`)

Reply:
Should we move the noncallable number one here as well then?

Reply:
I'm not entirely sure. On the one hand, we want the default experience to have good errors, even on CI. On the other hand, we don't want to encumber the runtime with complicated error messages that aren't needed, even on CI.

Reply:
Since these hint are really only intended for beginners, I would lean towards having them defined by the REPL. Users contributing to packages and looking at CI probably don't benefit much from them.